presto-product-tests/src/main/java/com/facebook/presto/tests/ImpersonationTests.java

```diff
@@ -0,0 +1,95 @@
+public class ImpersonationTests
+        extends ProductTest {
+    @Inject(optional = true)
+    private HdfsClient hdfsClient;
+
+    @Inject(optional = true)
+    @Named("databases.presto.jdbc_user")
+    private String prestoJdbcUser;
+
+    @Inject(optional = true)
+    @Named("databases.hive.warehouse_directory_path")
+    private String warehouseDirectoryPath;
+
+    private String getTableLocation(String tablename) {
+        requireNonNull(warehouseDirectoryPath, "databases.hive.warehouse_directory_path is null");
+        requireNonNull(tablename, "Input tablename is null");
+        return warehouseDirectoryPath + tablename;
+    }
+
+    private void checkTableOwner(String tablename, String tablepath, String expectedOwner) {
+        query(format("DROP TABLE IF EXISTS %s", tablename));
+        query(format("CREATE TABLE %s AS SELECT * FROM NATION", tablename));
+        String owner = hdfsClient.getOwner(tablepath);
+        assertEquals(owner, expectedOwner);
+        query(format("DROP TABLE IF EXISTS %s", tablename));
+    }
+
+    // Without impersonation, the table is created with user specified for
+    // -DHADOOP_USER_NAME in jvm.config
+    @Test(groups = {SINGLENODE, PROFILE_SPECIFIC_TESTS})
+    public void testOwnerSingleNode()
+            throws Exception {
+        String tableName = "check_owner_singlenode";
+        String tableLocation = getTableLocation(tableName);
+        checkTableOwner(tableName, tableLocation, "hive");
+    }
+
+    @Test(groups = {SINGLENODE_HDFS_IMPERSONATION, PROFILE_SPECIFIC_TESTS})
+    public void testOwnerSingleNodeHdfsImpersonation()
+            throws Exception {
+        String tableName = "check_owner_singlenode_hdfs_impersonation";
+        requireNonNull(prestoJdbcUser, "databases.presto.jdbc_user is null");
+        String tableLocation = getTableLocation(tableName);
+        checkTableOwner(tableName, tableLocation, prestoJdbcUser);
+    }
+
+    @Test(groups = {SINGLENODE_KERBEROS_HDFS_IMPERSONATION, PROFILE_SPECIFIC_TESTS})
+    public void testOwnerSingleNodeKerberosHdfsImpersonation()
+            throws Exception {
+        String tableName = "check_owner_singlenode_kerberos_hdfs_impersonation";
+        requireNonNull(prestoJdbcUser, "databases.presto.jdbc_user is null");
+        String tableLocation = getTableLocation(tableName);
+        checkTableOwner(tableName, tableLocation, prestoJdbcUser);
+    }
+
+    // Without hdfs impersonation, the table should be created with user specified for the
+    // principal hive.hdfs.presto.principal in hive.properties
+    @Test(groups = {SINGLENODE_KERBEROS_HDFS_NO_IMPERSONATION, PROFILE_SPECIFIC_TESTS})
+    public void testOwnerSingleNodeKerberosHdfsNoImpersonation()
+            throws Exception {
+        String tableName = "check_owner_singlenode_kerberos_hdfs_no_impersonation";
+        String tableLocation = getTableLocation(tableName);
+        checkTableOwner(tableName, tableLocation, "hdfs");
```

Comment:
Instead of hardcoding user names maybe it is better to have a property for this, say `databases.presto.configured_hdfs_user`? Or maybe we should change the approach how do we test that impersonation is disabled? We can create 2 different tables from 2 different JDBC users in Presto, and than check that the owner is the same for those 2 tables. I don't know which approach is better.